File: ruby/leet_code/1544.rb
# @param {String} s
# @return {String}
def make_good(s)
  stack = []
  s.each_char do |c|
    if stack.empty?
      stack.push(c)
    else
      if stack.last.downcase == c.downcase && stack.last != c
        stack.pop
      else
        stack.push(c)
      end
    end
  end
  stack.join
end

File: javascript/leet_code/1544.js
/**
 * @param {string} s
 * @return {string}
 */
let makeGood = function (s) {
    let stack = [];
    for (let i = 0; i < s.length; i++) {
        if (0 === stack.length) {
            stack.push(s[i]);
        } else {
            let top = stack[stack.length - 1];
            if (top.toLowerCase() === s[i].toLowerCase() && top !== s[i]) {
                stack.pop();
            } else {
                stack.push(s[i]);
            }
        }
    }
    return stack.join("");
};
